IaC/JenkinsImagesECR.yml

```
@@ -0,0 +1,134 @@
+        Architectures:
+          - "x86"
+          - "x86-64"
+
+  PXCBuildRepo:
+    Type: AWS::ECR::PublicRepository
+    Properties:
+      RepositoryName: "pxc-build"
+      RepositoryCatalogData:
+        UsageText: "This image used exclusively for Percona Jenkins pipelines"
+        AboutText: "This image contains all build utilities to create proper builds and execute tests"
+        OperatingSystems:
+          - "Linux"
+        Architectures:
+          - "x86"
+          - "x86-64"
+
+  ECRRWUser:
+    Type: AWS::IAM::User
+    Properties: 
+      Policies: 
+      - PolicyName: PushImagesECR
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Resource:
+            - !GetAtt PSBuildRepo.Arn
+            - !GetAtt PXCBuildRepo.Arn            
+            Action:
+            - ecr-public:BatchCheckLayerAvailability
+            - ecr-public:GetRepositoryPolicy
+            - ecr-public:DescribeRepositories
+            - ecr-public:DescribeRegistries
+            - ecr-public:DescribeImages
+            - ecr-public:DescribeImageTags
+            - ecr-public:GetRepositoryCatalogData
+            - ecr-public:GetRegistryCatalogData
+            - ecr-public:BatchCheckLayerAvailability
+            - ecr-public:PutImage
+            - ecr-public:InitiateLayerUpload
+            - ecr-public:UploadLayerPart
+            - ecr-public:CompleteLayerUpload
+      - PolicyName: GetAuthToken
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Resource:
+            - '*'    
+            Action:
+            - ecr-public:GetAuthorizationToken
+            - sts:GetServiceBearerToken
+      Tags: 
+        - Key: iit-billing-tag
+          Value: jenkins-ecr
+      UserName: "ECRRWUser"
+
+  ECRROUser:
+    Type: AWS::IAM::User
+    Properties: 
+      Policies: 
+      - PolicyName: PullImagesECR
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Resource:
+            - !GetAtt PSBuildRepo.Arn
+            - !GetAtt PXCBuildRepo.Arn            
+            Action:
+            - ecr-public:BatchCheckLayerAvailability
+            - ecr-public:GetRepositoryPolicy
+            - ecr-public:DescribeRepositories
+            - ecr-public:DescribeRegistries
+            - ecr-public:DescribeImages
+            - ecr-public:DescribeImageTags
+            - ecr-public:GetRepositoryCatalogData
+            - ecr-public:GetRegistryCatalogData
+      - PolicyName: GetAuthToken
+        PolicyDocument:
+          Version: 2012-10-17
+          Statement:
+          - Effect: Allow
+            Resource:
+            - '*'    
+            Action:
+            - ecr-public:GetAuthorizationToken
+            - sts:GetServiceBearerToken
+      Tags: 
+        - Key: iit-billing-tag
+          Value: jenkins-ecr
+      UserName: "ECRROUser"
```

please it is critical to be able to identify resources from the name
```suggestion
          Value: jenkins-ps-ecr
      UserName: "jenkins-ps-ecr"
```